src/GuestList/utils.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { Button, Col, Container, Row } from 'react-bootstrap';
+import mockData from './mockData';
+
+export const editGuestInfo = (e) => {
+  const guestInfo = e.currentTarget.nextSibling;
+
+  return guestInfo.className.indexOf('collapse show') > -1
+    ? guestInfo.classList.remove('show')
+    : guestInfo.classList.add('show');
+};
+
+export const readNoteFromGuest = (e) => {
+  const currentRow = e.currentTarget.closest('tr');
+  const noteFromGuest = currentRow.nextSibling.nextSibling;
+
+  return noteFromGuest.className.indexOf('collapse show-message') > -1
+    ? noteFromGuest.classList.remove('show-message')
+    : noteFromGuest.classList.add('show-message');
+};
+
+export const createListOfGuests = () => {
+  const guestData = Object.keys(mockData.guests);
+  const guestList = guestData.map((entry) => {
+    const guests = mockData.guests[entry].map((guest) => (
+      <>
+        <tr onClick={editGuestInfo}>
+          <td />
+          <td>{guest.firstName + guest.lastName}</td>
+          <td>{guest.email}</td>
+          <td>{guest.rsvpStatus}</td>
+          {guest.rsvpNote ? (
+            <td>
+              <Button onClick={readNoteFromGuest}>Note</Button>
+            </td>
+          ) : (
+            <td />
+          )}
+        </tr>
+        <tr className="collapse">
+          <td />
+          <td>Edit Name</td>
+          <td>Edit Email</td>
+          <td colSpan="2">
+            <Button>Delete Guest</Button>
+          </td>
+        </tr>
+        {guest.rsvpNote ? (
+          <tr className="collapse show-message">
+            <td colSpan="5">{guest.rsvpNote}</td>
+          </tr>
+        ) : null}
+      </>
+    ));
+    return guests;
+  });
+  return guestList;
+};
+
+export const createRsvpStats = () => {
+  const guestData = Object.keys(mockData.guests);
+  const attendance = {
+    attending: 0,
+    declined: 0,
+    pending: 0,
+  };
+
+  guestData.forEach((entry) => {
+    attendance.attending += mockData.guests[entry].filter(
+      (guest) => guest.rsvpStatus === 'attending'
+    ).length;
+  });
+  guestData.forEach((entry) => {
+    attendance.pending += mockData.guests[entry].filter(
+      (guest) => guest.rsvpStatus === 'pending'
+    ).length;
+  });
+  guestData.forEach((entry) => {
+    attendance.declined += mockData.guests[entry].filter(
+      (guest) => guest.rsvpStatus === 'declined'
+    ).length;
+  });
```

Comment:
probably a way to iterate through guestdata once to get these values like two/three different if statements, tradeoff would probably not using filter but might be worth it to reduce time complexity